base/gcutils.jl

```diff
@@ -119,9 +119,9 @@ const GC_INCREMENTAL = 2
     GC.gc([full=true])
 
 Perform garbage collection. The argument `full` determines the kind of
-collection: A full collection (default) sweeps all objects, which makes the
-next GC scan much slower, while an incremental collection may only sweep
-so-called young objects.
+collection: a full collection (default) traverses all live objects (i.e. full mark)
+and should reclaim memory from all unreachable objects. An incremental collection only
+reclaims memory from young objects which are not reachable.
 
```

Comment:
```suggestion

The GC may decide to perform a full collection even if an incremental collection was requested.

```

Reply:
Actually, not sure if this is true?

Reply:
This is true. If the last sweep happened, by chance, to be full, then we'll do a full mark.